cql3/statements/create_keyspace_statement.cc

```diff
@@ -94,21 +94,42 @@ future<std::tuple<::shared_ptr<cql_transport::event::schema_change>, std::vector
     const auto& feat = qp.proxy().features();
     ::shared_ptr<event::schema_change> ret;
     std::vector<mutation> m;
+    std::vector<sstring> warnings;
 
     try {
-        m = service::prepare_new_keyspace_announcement(qp.db().real_database(), _attrs->as_ks_metadata(_name, tm, feat), ts);
+        auto ksm = _attrs->as_ks_metadata(_name, tm, feat);
+        m = service::prepare_new_keyspace_announcement(qp.db().real_database(), ksm, ts);
 
         ret = ::make_shared<event::schema_change>(
                 event::schema_change::change_type::CREATED,
                 event::schema_change::target_type::KEYSPACE,
                 keyspace());
+
+        // If the new keyspace uses tablets, as long as there are features
+        // which aren't supported by tablets we want to warn the user that
+        // they will not be usable on the new keyspace - and suggest how a
+        // keyspace can be created without tablets (see rationale in #16807).
+        // Once all feature will become supported with tablets, we should
+        // remove this check.
+        auto rs = locator::abstract_replication_strategy::create_replication_strategy(
+            ksm->strategy_name(),
+            locator::replication_strategy_params(ksm->strategy_options(), ksm->initial_tablets()));
+        if (rs->uses_tablets()) {
+            warnings.push_back(
+                "Tables in this keyspace will be replicated using tablets, "
+                "and will not support the CDC feature (issue #16317) and LWT "
```

Comment:
CDC & LWT are mentioned explicitly, but aren't/won't there be any other such features? Maybe we can generalize this statement.

Reply:
I think this message will be pointless if it doesn't explicitly list the specific features that have problems with tablets. I don't want to "cover my ass" and say something like "some things don't work with tablets, but I'm not going to commit on what doesn't work, so just watch your step". 
If we discover more problems, we should update this message.